src/main/java/com/example/onlinebookstore/validation/FieldMatchValidator.java

```diff
@@ -0,0 +1,29 @@
+package com.example.onlinebookstore.validation;
+
+import com.example.onlinebookstore.dto.UserRequestDto;
+import com.example.onlinebookstore.exception.RegistrationException;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.lang.reflect.Field;
+import java.util.Objects;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, UserRequestDto> {
+    @Override
+    public boolean isValid(UserRequestDto userRequestDto,
+                           ConstraintValidatorContext constraintValidatorContext) {
+        Object field1Value = getFieldValue(userRequestDto, "password");
+        Object field2Value = getFieldValue(userRequestDto, "repeatPassword");
+
+        return Objects.equals(field1Value, field2Value);
+    }
+
+    private Object getFieldValue(UserRequestDto userRequestDto, String fieldName) {
+        try {
+            Field field = UserRequestDto.class.getDeclaredField(fieldName);
+            field.setAccessible(true);
+            return field.get(userRequestDto);
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            throw new RegistrationException("can't find the appropriate field");
```

Comment:
* I wouldn't say that `RegistrationException` is the appropriate use here, we can use the same validator for user password update endpoint
* make ex message more informative if you have such an opportunity
```suggestion
            throw new IllegalArgumentException("can't find the appropriate field: " + fieldName);
```